### ISSUE TITLE
getAll queries should be able to search by more than email.

### ISSUE THREAD

**Issue by aljones15** [digitalbazaar/bedrock-account-http#16]
The google Docs for getAll specified only search by email.
@mattcollier @gannan08 both want to expand this search functionality.
even the wireframe expects search by more than just email.
so we could refactor:

1. return the email check to exists query and to the current getAll query
2. allow searches by account.meta created,
3. allow searches by status (and filter by status)
4. allow search by sysResourceRole

in the future we could expand to search by credentials / persona
this would allow search by phone number, last name, first name, etc.

**Comment by mattcollier** [comment]
Right, so on the topic of `meta.status`, the http API provides no ability to restrict query by status, and the `getAll` API in bedrock-account has no defaults related to status (unlike the exists API).  Therefore, all requests to the http API will include account with any status without any way to affect that behavior.

https://github.com/digitalbazaar/bedrock-account/blob/master/lib/index.js#L275
